setup.py

```diff
@@ -182,23 +140,37 @@ def members_appended(*ds):
         print('Warning: can not import pkgconfig python package.')
         pc = None
 
+    def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version):
+        system_prefix = os.environ.get(prefix_env_var)
+        if system_prefix:
+            print(f'Detected and preferring {lib_pkg_name} [via {prefix_env_var}]')
+            return dict(include_dirs=[os.path.join(system_prefix, 'include')],
+                        library_dirs=[os.path.join(system_prefix, 'lib')],
+                        libraries=[lib_name])
+
+        if pc and pc.installed(lib_pkg_name, pc_version):
+            print(f'Detected and preferring {lib_pkg_name} [via pkg-config]')
+            return pc.parse(lib_pkg_name)
+        raise Exception(f'Could not find {lib_name} lib/headers, please set {prefix_env_var}')
+
+
     crypto_ext_kwargs = members_appended(
         dict(sources=[crypto_ll_source, crypto_helpers]),
-        setup_crypto.crypto_ext_kwargs(pc, system_prefix_openssl),
+        lib_ext_kwargs(pc, 'BORG_OPENSSL_PREFIX', 'crypto', 'libcrypto', '>=1.1.1'),
```

Comment:
Please note that we also support libressl and i suspect its libcrypto won't have a openssl compatible version number?

That is why the orginal code didn't check for the version, but just for existence.

We have checks in the borg code that makes sure the crypto is not too old.

Reply:
Maybe some always true version check could also work, like `>=0` or so?

Reply:
Just [checked](http://www.libressl.org/releases.html) and LibreSSL started at version 2.0, so will always be higher than openssl.